src/components/templates/descopeMiddleware.tsx

```diff
@@ -238,6 +260,7 @@ export const DescopeMiddleware = ({ children }: { children: ReactNode }) => {
 
 	const isLoggedIn = user && Cookies.get(systemCookies.isLoggedIn);
 	if ((playwrightTestsAuthBearer || apiToken || isLoggedIn) && !isLoggingIn) {
+		handleCliLoginRedirect();
```

Comment:
The render phase calls `handleCliLoginRedirect()` but ignores its return value, causing the component to render `children` even when a redirect is triggered.
This way we get a race condition: `handleCliLoginRedirect()` sets `window.location.href` but immediately returns, allowing React to render children before the browser navigates. This causes a UI flash where users briefly see the app content. The return value is ignored, so we can't prevent rendering during the redirect.